### ISSUE TITLE
Consider saturated arithmetic functionality for Vector<T>

### ISSUE THREAD

**Issue by mellinoe** [dotnet/runtime#14280]
In the past, we've seen some requests for Vector<T> to provide some means to do saturated arithmetic operations. Such operations could be useful for a variety of applications, like signal processing, image processing, etc. It makes sense to at least investigate what such functionality would look like, how it could be exposed, etc. Given that there are also SSE2+ instructions that expose some level of saturated arithmetic, I'm hopeful that we could also get some baseline hardware-instruction support for this.

Work to do:
- Define API surface exposed (which operations are supported)
- Define which types support these operations (which integral types? all?)
- Determine which operations could benefit from SIMD instructions
- Create a simple working prototype (without hardware instructions)
- Create formal API design speclet

**Comment by tannergooding** [comment]
Going to close this for now. This hasn't gained much traction and likely needs more thought/scenarios before it could be moved forward.

If someone feels strongly about this, please feel free to open a new issue. Our API review process is detailed here with a "good example" being listed under step 1. Ideally the new issue would provide many of the sections given, but would at a minimum provide the base rationale and proposed API surface sections.

For this API in particular, the exposure of the System.Runtime.Intrinsic types and the methods for converting to/from `Vector<T>` should unblock users who immediately need this functionality.
